runtime/near-vm-logic/src/logic.rs

```diff
@@ -2191,7 +2192,9 @@ impl<'a> VMLogic<'a> {
         let buf = self.get_vec_from_memory_or_register(ptr, len)?;
         self.gas_counter.pay_base(utf8_decoding_base)?;
         self.gas_counter.pay_per(utf8_decoding_byte, buf.len() as u64)?;
-        let account_id = AccountId::from_utf8(buf).map_err(|_| HostError::BadUTF8)?;
+        let account_id =
+            AccountId::try_from(String::from_utf8(buf).map_err(|_| HostError::BadUTF8)?)
+                .map_err(|_| HostError::InvalidAccountId)?;
```

Comment:
This still changes the protocol. We now return an error that we won't return previously if the account id parsed is 65 bytes for example. cc @frol

Reply:
@bowenwang1996 @olonho I don't understand whether an account id read by this method can be invalid or it somehow maintains the invariant of being valid?

Reply:
If the account id is always valid, we may as well use `.expect` here or abuse `BadUTF8` error variant (that would be unfortunate).

If the account id is not always valid, we may either abuse `BadUTF8` (again, unfortunate) or use `String` instead of `AccountId`.

Reply:
> If the account id is not always valid, we may either abuse BadUTF8 (again, unfortunate) or use String instead of AccountId.

That is not the problem. The problem here is that this function could successfully parse an account id that is 65 characters long today and with this change it won't be able to and will return some error (does not matter which one it returns), which caused the behavior of the protocol to change.

Reply:
@bowenwang1996 I understand the problem of a protocol-breaking change very well; sorry for not being explicit about that. I am trying to understand if we want to break it or it is fine to just return `String` here and no breaking change will occur.

Reply:
> I am trying to understand if we want to break it or it is fine to just return String here and no breaking change will occur.

Good question. Ideally it should return a valid account id, but it is not urgent, so maybe let's just return a `String` for now

Reply:
We (@miraclx and I) reviewed the code and arrived to a non-breaking solution.

1. We decided to introduce an "unsafe" constructor to `AccountId` type, which lets to instantiate it from any string. This is brutal and does not exactly holds the contract of `AccountId` instance always holding a valid account id; however, after cosidering the tradeoffs, we decided to go this path.
2. We manually reviewed the code path that is triggered and learned that the constructed value only lives for a certain time before it gets finally validated (the "unsafe" behaviour is completely safe today with the String type alias, and remains to be safe with this PR)
3. We hid the "unsafe" constructor befind a feature flag, so hopefully nobody else would need it
4. There should be no breaking change